backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,252 @@
+                List<Calendar> calendars = calendarManager.getCalendars();
+                response.setResponseBody(JsonContent.writableOf(calendars))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "getAppointments")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointments(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Appointment> appointments = appointmentManager.readAllBy(userId.get(), null, null);
+                response.setResponseBody(JsonContent.writableOf(appointments))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "getAppointmentsByDate")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointmentsByDate(RequestContext context, Date from, Date to) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Appointment> appointments = appointmentManager.readAllBy(userId.get(), from, to);
+                response.setResponseBody(JsonContent.writableOf(appointments))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "getAppointment")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointment(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                request.getRequestBody()
+                        .flatMap(body -> body.tryRead(Appointment.class))
+                        .flatMap(appointment -> {
+                            Appointment foundAppointment = appointmentManager.read(appointment.getCalendarId(), appointment.getId());
```

Comment:
Why do we have an appointment body here, if we want to get an existing appointment?